app/services/billable_metrics/advanced_aggregations/prorated_unique_count_service.rb

```diff
@@ -0,0 +1,177 @@
+      end
+
+      def prorated_removed_query
+        base_scope
+          .where('quantified_events.added_at::timestamp(0) < ?', from_datetime)
+          .where('quantified_events.removed_at::timestamp(0) >= ?', from_datetime)
+          .where('quantified_events.removed_at::timestamp(0) <= ?', to_datetime)
+      end
+
+      def prorated_added_and_removed_query
+        base_scope
+          .where('quantified_events.added_at::timestamp(0) >= ?', from_datetime)
+          .where('quantified_events.added_at::timestamp(0) <= ?', to_datetime)
+          .where('quantified_events.removed_at::timestamp(0) >= ?', from_datetime)
+          .where('quantified_events.removed_at::timestamp(0) <= ?', to_datetime)
+      end
+
+      def base_scope
+        quantified_events = QuantifiedEvent
+          .joins(customer: :organization)
+          .where(billable_metric_id: billable_metric.id)
+          .where(customer_id: subscription.customer_id)
+          .where(external_subscription_id: subscription.external_id)
+
+        return quantified_events unless group
+
+        group_scope(quantified_events)
+      end
+
+      # NOTE: Compute pro-rata of the duration in days between the datetimes over the duration of the billing period
+      #       Dates are in customer timezone to make sure the duration is good
+      def duration_ratio_sql(from, to)
+        from_in_timezone = Utils::TimezoneService.date_in_customer_timezone_sql(customer, from)
+        to_in_timezone = Utils::TimezoneService.date_in_customer_timezone_sql(customer, to)
+
+        "SUM((DATE(#{to_in_timezone}) - DATE(#{from_in_timezone}) + 1)::numeric / #{period_duration})::numeric"
+      end
+
+      def extend_event_metadata(prorated_value)
+        unless previous_event
+          result.max_aggregation_with_proration = prorated_value.to_s
+
+          return
+        end
+
+        if BigDecimal(result.max_aggregation) > BigDecimal(previous_event.metadata['max_aggregation'])
+          result.max_aggregation_with_proration =
+            (BigDecimal(previous_event.metadata['max_aggregation_with_proration']) + prorated_value).to_s
+        else
+          result.max_aggregation_with_proration = BigDecimal(previous_event.metadata['max_aggregation_with_proration'])
+        end
+      end
+
+      def handle_current_usage(result_without_proration, result_with_proration, is_pay_in_advance)
+        value_without_proration = result_without_proration.aggregation
+
+        if !is_pay_in_advance
+          result.aggregation = result_with_proration.negative? ? 0 : result_with_proration
+          result.current_usage_units = value_without_proration.negative? ? 0 : value_without_proration
+        elsif previous_event
+          result.current_usage_units = result_without_proration.current_usage_units
+          result.aggregation = result_without_proration.current_usage_units -
+            BigDecimal(previous_event.metadata['current_aggregation']) +
```

Comment:
Layout/MultilineOperationIndentation: Align the operands of an expression in an assignment spanning multiple lines.